spec/graphql/types/albums/album_search_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'rails_helper'
+require 'support/graphql/fixtures/graphql_album_helper'
+
+module Types
+  module Albums
+    RSpec.describe QueryType, type: :request do
+      describe '.resolve' do
+        context 'when get a album by id' do
+          before do
+            album_search_request_success
+            photos_for_album_success
+            post '/graphql', params: { query: query_get_album(id: '1') }
+          end
+
+          it 'return the id album' do
+            expect(JSON.parse(response.body)['data']['album']['id']).to eql('1')
+          end
+
+          it 'return the title album' do
+            expect(JSON.parse(response.body)['data']['album']['title'])
+              .to eql('delectus aut autem')
+          end
+
+          it 'return the userId album' do
+            expect(JSON.parse(response.body)['data']['album']['userId'])
+              .to eql('1')
+          end
+
+          it 'return the count of photos' do
+            expect(JSON.parse(response.body)['data']['album']['photos'].count)
+              .to eq(50)
+          end
+
+          it 'return the first photos' do
+            (expect(JSON.parse(response.body)['data']['album']['photos'].first['id'])
```

Comment:
Maybe you can check directly that the object is the same as what you are expecting instead of doing it per field.

Reply:
For instance, instead of doing this you can build a hash object that has the same structure as the response.body that you are expecting and check if both are the same, ie:

```
    it 'returns the first photo' do
      expected = {
        id: 1,
        title: 'A photo',
        url: 'google.com',
        thumbnailUrl: 'google.com',
      }

      expect(result).to eq expected
    end
```